packages/core/src/browser/keybinding.ts

```diff
@@ -622,7 +628,7 @@ export class KeybindingRegistry {
         this.keySequence.push(keyCode);
         const bindings = this.getKeybindingsForKeySequence(this.keySequence);
 
-        if (this.tryKeybindingExecution(bindings.full, event)) {
+        if (bindings.partial.length === 0 && this.tryKeybindingExecution(bindings.full, event)) {
```

Comment:
Are you sure that it is correct? If we already collected the full keybinding we should execute it, it does not matter whether there are other not matched keybindings yet.

I think it should be reverted. What case have you tried to fix?

Reply:
> it does not matter whether there are other not matched keybindings yet.
They are matched, `bindings.partial` contain keybindings that consist of several key combinations e.g. `Ctrl-x` and `Ctrl-l`. If `bindings` contains such keybinding in the `partial` list it would be ignored because the full `Ctrl-x` keybinding is registered.
May be it would be better to place this check lower, after the `bindings.partial` check ?

Reply:
If  `Ctrl-x` is registered then it should be executed, it does not matter that there is something for `Ctrl-x Ctrl-l`. It is an issue in keybinding configuration. You can try it is the same in VS Code.

In which context it should work? It could be a bug in `when` or `context` properties for `Ctrl-x`.

Reply:
VsCode doesn't separate key bindings to full, partial etc. It collects matched keybindings into a single collection and takes the first matched given context from the bottom of the list.
Theia has three key binding types: full, partial and shadow. The key bindings like `Ctrl-x Ctrl-l` are listed in the `partial` list, so if `Ctrl-x` is registered, it will be listed in the `full` list which is checked first.

Reply:
I would need to test again, cannot do it now. But according to your comment for ctrl+x one keybinding already will match, no? do ctrl+l will be handled independently. the whole sequence only will match ignoring matched to ctrl+f

Reply:
> one keybinding already will match

Yes, `Ctrl-x` matches the context, but `Ctrl-x + Ctrl-l` matches as well.
> `ctrl+l` will be handled independently

It can not be handled without `ctrl-x`. Once `Ctrl-x` is pressed, and if such duuble key binding is registered, IDE listens to the second key combination, a message about it is shown in the status bar.

Reply:
ok, tested it with VS Code, it seems you are right.

We don't have many kebindings using combinations, so it probably should not break. @svenefftinge @marechal-p sounds good?

Reply:
LGTM